fixtures/chauvet-dj/megastrobe-fx12.json

```diff
@@ -0,0 +1,307 @@
+          "comment": "Synchronous"
+        },
+        {
+          "dmxRange": [186, 202],
+          "type": "ShutterStrobe",
+          "shutterEffect": "RampDown",
+          "comment": "Synchronous"
+        },
+        {
+          "dmxRange": [203, 219],
+          "type": "ShutterStrobe",
+          "shutterEffect": "RampUpDown",
+          "comment": "Synchronous"
+        },
+        {
+          "dmxRange": [220, 236],
+          "type": "ShutterStrobe",
+          "shutterEffect": "Strobe",
+          "comment": "Synchronous"
+        },
+        {
+          "dmxRange": [237, 255],
+          "type": "ShutterStrobe",
+          "shutterEffect": "Closed"
+        }
+      ]
+    },
+    "Shutter Speed": {
+      "capability": {
+        "type": "Generic",
+        "comment": "Sensitivity OR Shutter Speed"
+      }
+    },
+    "Pixels": {
+      "capability": {
+        "type": "Generic",
+        "comment": "Pixel, Program Speed, or Sound Active Auto"
+      }
+    },
+    "Auto Programs": {
+      "capabilities": [
+        {
+          "dmxRange": [0, 48],
+          "type": "NoFunction"
+        },
+        {
+          "dmxRange": [49, 64],
+          "type": "Effect",
+          "effectName": "Auto Program 1"
+        },
+        {
+          "dmxRange": [65, 80],
+          "type": "Effect",
+          "effectName": "Auto Program 2 "
+        },
+        {
+          "dmxRange": [81, 96],
+          "type": "Effect",
+          "effectName": "Auto Program 3"
+        },
+        {
+          "dmxRange": [97, 112],
+          "type": "Effect",
+          "effectName": "Auto Program 4"
+        },
+        {
+          "dmxRange": [113, 128],
+          "type": "Effect",
+          "effectName": "Auto Program 5 "
+        },
+        {
+          "dmxRange": [129, 144],
+          "type": "Effect",
+          "effectName": "Auto Program 6"
+        },
+        {
+          "dmxRange": [145, 160],
+          "type": "Effect",
+          "effectName": "Auto Program 7"
+        },
+        {
+          "dmxRange": [161, 176],
+          "type": "Effect",
+          "effectName": "Auto Program 8"
+        },
+        {
+          "dmxRange": [177, 192],
+          "type": "Effect",
+          "effectName": "Auto Program 9"
+        },
+        {
+          "dmxRange": [193, 208],
+          "type": "Effect",
+          "effectName": "Auto Program 10"
+        },
+        {
+          "dmxRange": [209, 224],
+          "type": "Effect",
+          "effectName": "Auto Program 11"
+        },
+        {
+          "dmxRange": [225, 240],
+          "type": "Effect",
+          "effectName": "Auto Program 12"
+        },
+        {
+          "dmxRange": [241, 255],
+          "type": "Effect",
+          "effectName": "Sound Active",
+          "soundControlled": true
+        }
+      ]
+    },
+    "Dimmer 1": {
+      "capability": {
+        "type": "Intensity"
+      }
+    },
+    "Dimmer 2": {
```

Comment:
Can these extra dimmers be consolidated in any way?

Reply:
Yes, with matrices: https://github.com/OpenLightingProject/open-fixture-library/blob/master/docs/fixture-format.md#matrices

Reply:
Ok, I think I've implemented the matrix correctly. I did notice that there is an `all` key which could be used for the master dimmer, but then I'm not sure how to exclude it from the `eachPixelGroup` computation.

Reply:
@FloEdelmann I've tried but it appears I don't really understand the matrix definitions well enough. I've tried to follow examples from other fixtures but I'm a bit stuck. Also, for whatever reason the tests won't run locally on my machine, apologies for the messy commit history!

Reply:
Since there are two axes, the implicit pixel keys are in the format `x,y`, e.g. `1,1`, `1,2`, `2,1`, …

I think it would be better to be explicit here. Instead of specifying the `pixelCount`, better specify the `pixelKeys`:

```json5
"matrix": {
  "pixelKeys": [
    [
      ["LED 1", "LED 2", "LED 3", "LED 4", "LED 5", "LED 6"],
      ["LED 7", "LED 8", "LED 9", "LED 10", "LED 11", "LED 12"]
    ]
  ],
  "pixelGroups": {
    "Group 1": ["LED 1", "LED 7"],
    // …
  }
}
```

Reply:
> there is an `all` key which could be used for the master dimmer, but then I'm not sure how to exclude it from the `eachPixelGroup` computation

You can't, but you could list the pixel groups manually (as an array) in the channel list.